services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -486,6 +489,168 @@ async def list_projects(  # pylint: disable=too-many-arguments
+                    (
+                        (projects.c.prj_owner == user_id)
+                        | sa.text(
+                            f"jsonb_exists_any(access_rights_subquery.access_rights, {assemble_array_groups(user_groups)})"
+                        )
+                    )
+                    & (projects.c.workspace_id.is_(None))
+                    & (projects_to_products.c.product_name == product_name)
+                    & (projects.c.hidden.is_(False))
+                    & (projects.c.type == ProjectType.STANDARD)
+                    & (
+                        (projects.c.name.ilike(f"%{text}%"))
+                        | (projects.c.description.ilike(f"%{text}%"))
+                        | (projects.c.uuid.ilike(f"%{text}%"))
+                    )
+                )
+            )
+
+            shared_workspace_query = (
+                sa.select(
+                    *[
+                        col
+                        for col in projects.columns
+                        if col.name not in ["access_rights"]
+                    ],
+                    workspace_access_rights_subquery.c.access_rights,
+                    projects_to_products.c.product_name,
+                    projects_to_folders.c.folder_id,
+                )
+                .select_from(
+                    projects.join(
+                        workspace_access_rights_subquery,
+                        projects.c.workspace_id
+                        == workspace_access_rights_subquery.c.workspace_id,
+                    )
+                    .join(projects_to_products)
+                    .join(
+                        projects_to_folders,
+                        (
+                            (projects_to_folders.c.project_uuid == projects.c.uuid)
+                            & (projects_to_folders.c.user_id == user_id)
+                        ),
+                        isouter=True,
+                    )
+                )
+                .where(
+                    (
+                        sa.text(
+                            f"jsonb_exists_any(workspace_access_rights_subquery.access_rights, {assemble_array_groups(user_groups)})"
+                        )
+                    )
+                    & (projects.c.workspace_id.is_not(None))
+                    & (projects_to_products.c.product_name == product_name)
+                    & (projects.c.hidden.is_(False))
+                    & (projects.c.type == ProjectType.STANDARD)
+                    & (
+                        (projects.c.name.ilike(f"%{text}%"))
+                        | (projects.c.description.ilike(f"%{text}%"))
+                        | (projects.c.uuid.ilike(f"%{text}%"))
+                    )
+                )
+            )
+
+            combined_query = sa.union_all(
+                private_workspace_query, shared_workspace_query
+            )
+
+            count_query = sa.select([func.count()]).select_from(combined_query)
```

Comment:
AFAIK this is incompatible with sqlalchemy 2.0 and should be printing some WARNINGs in your tests :-)

I think the right way is without the list `[.]`
```python
count_query = sa.select( func.count() ).select_from(combined_query)
```

Reply:
Tests are coming in a moment ;)